src/Widgets/VPN/VPNMenuItem.vala

```diff
@@ -16,106 +16,159 @@
+            if (event.detail == Gdk.NotifyType.INFERIOR) {
+                return false;
+            }
+            debug ("Exit");
 
-    /**
-     * Only used for an item which is not displayed: hacky way to have no radio button selected.
-     **/
-    public VPNMenuItem.blank () {
-        radio_button = new Gtk.RadioButton (null);
-    }
+            vpn_info_revealer.set_reveal_child (false);
+            return false;
+        });
 
-    public void set_active (bool active) {
-        radio_button.set_active (active);
+        add (event_box);
+        show_all ();
     }
 
     private void update () {
-        radio_button.label = connection.get_id ();
+        vpn_label.label = connection.get_id ();
 
         switch (state) {
-        case State.FAILED_VPN:
-            show_item (error_img);
-            break;
-        case State.CONNECTING_VPN:
-            show_item (spinner);
-            break;
-        default:
-            hide_icons ();
-            break;
+            case State.FAILED_VPN:
+                state_label_text = _("Failed");
+                vpn_state.icon_name = "user-busy";
+                connect_button.label = _("Connect");
+                connect_button.sensitive = true;
+                connect_button.get_style_context ().remove_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                break;
+            case State.CONNECTING_VPN:
+                state_label_text = _("Connecting");
+                vpn_state.icon_name = "user-away";
+                connect_button.sensitive = false;
+                connect_button.get_style_context ().remove_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                break;
+            case State.CONNECTED_VPN:
+                state_label_text = _("Connected");
+                vpn_state.icon_name = "user-available";
+                connect_button.label = _("Disconnect");
+                connect_button.sensitive = true;
+                connect_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                break;
+            case State.DISCONNECTED:
+                state_label_text = _("Disconnected");
+                vpn_state.icon_name = "user-offline";
+                connect_button.label = _("Connect");
+                connect_button.sensitive = true;
+                connect_button.get_style_context ().remove_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                break;
+            default:
+                connect_button.label = _("Connect");
+                connect_button.sensitive = true;
+                break;
         }
-    }
 
-    public void hide_icons (bool show_remove_button = true) {
-        hide_item (error_img);
-        hide_item (spinner);
-
-        if (!show_remove_button) {
-            hide_item (remove_button);
-        }
+        state_label.label = GLib.Markup.printf_escaped ("<span font_size='small'>%s</span>", state_label_text);
```

Comment:
It looks like Utils has a `to_string` method for this enum: https://github.com/elementary/switchboard-plug-network/blob/master/src/Utils.vala#L40

So it's not necessary to supply our own strings here. We can simply use `state.to_string ()`